### PR TITLE
Accommodations to server-side whiteboards

### PR DESCRIPTION
**Pull request by hero101**
- `updateWhiteboardContent` mutation removed to disable the update of multi-user Whiteboards
- handling `fetch` and `save` requests from the `whiteboard-collaboration-service`
- `whiteboardSaved` and `whiteboardContentUpdated` subscriptions removed

## Summary by CodeRabbit

- **New Features**
	- Introduced new methods for saving and fetching whiteboard content in the integration service and controller.
	- Added classes for handling input and output data related to save and fetch operations.

- **Bug Fixes**
	- Enhanced error handling in the whiteboard service to manage save and fetch operations more robustly.

- **Refactor**
	- Simplified method signatures and removed outdated subscription functionalities related to whiteboard events. 

- **Chores**
	- Updated exports to include new input and output data classes for improved module interface.

### CODE REVIEW

File: src/domain/common/whiteboard/whiteboard.module.ts [hunk 1]
```diff
@@ -15,7 +15,6 @@ import { StorageBucketModule } from '@domain/storage/storage-bucket/storage.buck
 import { ProfileDocumentsModule } from '@domain/profile-documents/profile.documents.module';
 import { LicenseEngineModule } from '@core/license-engine/license.engine.module';
 import { SubscriptionServiceModule } from '@services/subscriptions/subscription-service';
```
Comment (ccanos) [inline src/domain/common/whiteboard/whiteboard.module.ts]: SubscriptionServiceModule is not used anymore

File: src/domain/common/whiteboard/whiteboard.service.ts [hunk 2]
```diff
@@ -24,18 +22,13 @@ import { Whiteboard } from './whiteboard.entity';
 import { IWhiteboard } from './whiteboard.interface';
 import { CreateWhiteboardInput } from './dto/whiteboard.dto.create';
 import { UpdateWhiteboardInput } from './dto/whiteboard.dto.update';
-import { WHITEBOARD_CONTENT_UPDATE } from './events/event.names';
 import { LicenseEngineService } from '@core/license-engine/license.engine.service';
 import { LicensePrivilege } from '@common/enums/license.privilege';
 import { SubscriptionPublishService } from '@services/subscriptions/subscription-service';
```
Comment (ccanos) [inline src/domain/common/whiteboard/whiteboard.service.ts]: please remove this one

**Review by ccanos — CHANGES_REQUESTED** [review]
just a couple of unused variables

File: src/services/whiteboard-integration/whiteboard.integration.service.ts [hunk 3]
```diff
@@ -98,6 +108,44 @@ export class WhiteboardIntegrationService {
...
+        }
+      );
+      return new FetchOutputData(new FetchContentData(whiteboard.content));
+    } catch (e: any) {
+      this.logger.error(e?.message, e?.stack, LogContext.EXCALIDRAW_SERVER);
+      return new FetchOutputData(
+        new FetchErrorData(e?.message ?? JSON.stringify(e))
+      );
+    }
+  }
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/services/whiteboard-integration/whiteboard.integration.service.ts]: This service method is used from another microservice and serves only for integration purposes - the microservice is authorized to read the information

File: src/services/whiteboard-integration/whiteboard.integration.service.ts [hunk 4]
```diff
@@ -98,6 +108,44 @@ export class WhiteboardIntegrationService {
...
+      );
+    } catch (e: any) {
+      this.logger.error(e?.message, e?.stack, LogContext.EXCALIDRAW_SERVER);
+      return new SaveOutputData(
+        new SaveErrorData(e?.message ?? JSON.stringify(e))
+      );
+    }
+    // return success on successful save
+    return new SaveOutputData(new SaveContentData());
+  }
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/services/whiteboard-integration/whiteboard.integration.service.ts]: This service method is used from another microservice and serves only for integration purposes - the microservice is authorized to read the information

**Review by ccanos — APPROVED** [review]

**Comment by ccanos** [comment]
DO NOT merge until whiteboard-collaboration-service and client are ready

**Pull request merged**